### PR TITLE
Bug/vspc 207

### PR DESCRIPTION
**Pull request by swetalina-prusty25**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 
Staff side: There should be a search for images

...Put description here...
When going to "Images" on the staff side, right now it shows a table of all images. There should be a search to filter images. The search should search filename, name, and description for the given search term.
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 1]
```diff
@@ -250,4 +283,43 @@ public void removeCategory(IVSImage image, ImageCategory category) {
...
+     * Method to return the requested images
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param images     - This is the result list of search images.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List <IVSImage> getImagesForPagination(int pageNo, List <IVSImage> images, String searchTerm) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:313]: why does this method get a list of images?

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    	if (searchTerm != null && !searchTerm.trim().isEmpty()) {
+            images = imageService.search(searchTerm);
+    	}
+    	else {
+            images = imageService.getImages(1);
+    	}
+    	int pageNo;
+    	page = StringUtils.isEmpty(page) ? "1" : page;
+    	try {
+            pageNo = imageService.validatePageNumber(Integer.parseInt(page), images);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:47]: the page object that the repository returns when you search should have all the info you need (page number, total pages, etc). no reason to make extra calls to the db here.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    
+    @RequestMapping("/staff/images/search/full/{page}")
+	public String imageSeacrhDescription(@PathVariable(required = false) String page,
+			@RequestParam(value = "searchText", required = false) String searchTerm, Model model,
+			RedirectAttributes attributes) {
+    	List<IVSImage> images = null;
+    	if (searchTerm != null && !searchTerm.trim().isEmpty()) {
+            images = imageService.search(searchTerm);
+    	}
+    	else {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:41]: please reread our style guide and follow it

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    	try {
+            pageNo = imageService.validatePageNumber(Integer.parseInt(page), images);
+    	}
+    	catch (NumberFormatException numberFormatException) {
+            pageNo = 1;
+    	}
+    	model.addAttribute("totalPages", imageService.getTotalPages(images));
+        model.addAttribute("currentPageNumber", pageNo);
+        model.addAttribute("totalImageCount", images.size());
+        List<IVSImage> imageResults = imageService.getImagesForPagination(pageNo, images, searchTerm);
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:55]: why is there another call the service/db? it should only be called once.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java (hunk 1 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:313]: fixed

**Pull request closed without merging**